### ISSUE TITLE
oil binary can behave like echo, printf, test, seq, cat, etc.

### ISSUE THREAD

**Issue by andychu** [oilshell/oil#801]
It already has `osh` and `oil`.  This gives it some more `busybox`-like functionality

**Comment by andychu** [comment]
We also have `readlink` now

**Comment by andychu** [comment]
- dirname, basename, abspath, etc.
- argv

from #799

**Comment by glyh** [comment]
I imagine that if one day this is ported to rust, we can utilize the existing coreutils rewrite without wasting man power on it. XD

**Comment by andychu** [comment]
Yeah Oil's philosophy is generally that the tools should be separate from the shell.  They can **evolve** independently, including in Rust, etc.

"Shells Should Shell Out" : https://www.oilshell.org/blog/2021/01/philosophy-design.html#shells-should-shell-out

However there is some internal functionality we need, and might as well expose it.

Actually this might be a dupe of #684 

I would draw the line between I/O and text processing.   We shouldn't add non-portable syscalls and I/O to Oil, but something like `seq` or `dirname` makes a lot of sense
